framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -51,29 +79,60 @@ public Void visitAnnotation(AnnotationTree node, Void p) {
         }
 
         Element element = TreeInfo.symbol((JCTree) node.getAnnotationType());
-        if (!(element.toString().equals(ArrayLen.class.getName())
+
+        if (element.toString().equals(IntRange.class.getName())) {
+            if (args.size() == 2
```

Comment:
It wouldn't return false. It was just a guard in case. 

So do you mean that, in good coding practice, we shouldn't have guards if the case we are guarding would never happen?